src/grammar.rs

```diff
@@ -319,7 +361,9 @@ pub enum Rule<Pat> {
 
     Opt(Rc<Rule<Pat>>),
     RepeatMany(Rc<Rule<Pat>>, Option<Rc<Rule<Pat>>>),
+    RepeatManyTrailing(Rc<Rule<Pat>>, Rc<Rule<Pat>>),
```

Comment:
IMO the distinction should go in the `Option` in the cases above - either next to the `Rc<Rule<Pat>>` (in a tuple), or wrapping it.

You can use the suggested names here for an `enum`: https://github.com/rust-lang-nursery/gll/pull/2/files#r264797019.